lib/api/core/models/repositories/profileRepository.js

```diff
@@ -130,7 +130,8 @@ module.exports = function (kuzzle) {
         // Fail if not all roles are found
         var rolesNotFound = _.difference(data.roles, extractRoleIds(roles));
         if (rolesNotFound.length) {
-          deferred.reject(new NotFoundError('Unable to hydrate profile with roles'));
+          deferred.reject(new NotFoundError(`Unable to hydrate the profile ${requestObject.data._id}. The following roles don't exist: ${rolesNotFound}`));
```

Comment:
requestObject is not defined here....
maybe you wanted to write 

    deferred.reject(new NotFoundError(`Unable to hydrate the profile ${requestObject.data._id}. The following roles don't exist: ${rolesNotFound}`));

?